src/main/java/duke/Task.java

```diff
@@ -0,0 +1,72 @@
+package duke;
+
+/**
+ * Represents a task that only contains a start date.
+ */
+public abstract class Task {
+  private String taskName;
+
+  private boolean isCompleted;
+
+  /**
+   * Creates a new Task with the given task name.
+   *
+   * @param taskName the name of the task
+   */
+  public Task(String taskName) {
+    this.taskName = taskName;
+  }
+
+  /**
+   * Creates a new Task with the given task name.
+   *
+   * @param taskName the name of the task
+   * @param isCompleted the completion status of the task
+   */
+  public Task(String taskName, boolean isCompleted) {
+    this.taskName = taskName;
+    this.isCompleted = isCompleted;
+  }
+
+  public abstract String saveData();
```

Comment:
Perhaps can name this method as `getSaveData` instead of `saveData` to make it clearer